### DISCUSSION TITLE
Register Plugin on ready

### DISCUSSION THREAD

**Discussion by meytin1337** [katspaugh/wavesurfer.js#3207] [Q&A]
Hello,
I am trying to register a plugin in an event callback but it does not seem to work.
I need access to the decoded audio data in the plugin options and that is not available before the "ready" event afaik.
Any ideas how to achieve this?
I know that an option would probably be to decode the audio data with another library but maybe there is an easier way?
This is what my code currently looks like:

```  
wavesurfer.on("ready", async () => {
    wavesurfer.setVolume(deck.volume);
    wavesurfer.zoom(deck.zoom);
    deck.isWaveformReady = true;
    deck.bpm = await analyze(wavesurfer.getDecodedData());
    wavesurfer.registerPlugin(
      TimelinePlugin.create({
        timeInterval: deck.bpm / (60 * 4),
      }),
    );
  });

```

**Comment by katspaugh** [comment]
Works for me. Try debugging your code.
